### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by cs-pranali-chavan**
- changes done to embed world map in another widget

### CODE REVIEW

File: widget/edit.controller.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    .module('cybersponse')
+    .controller('editCyberThreatWorldMap100Ctrl', editCyberThreatWorldMap100Ctrl);
+
+  editCyberThreatWorldMap100Ctrl.$inject = ['$scope', '$uibModalInstance', 'config', 'widgetUtilityService', '$timeout', 'modelMetadatasService', 'Entity'];
+
+  function editCyberThreatWorldMap100Ctrl($scope, $uibModalInstance, config, widgetUtilityService, $timeout, modelMetadatasService, Entity) {
+    $scope.cancel = cancel;
+    $scope.save = save;
+    $scope.config = config;
+    $scope.loadFieldsOfModule = loadFieldsOfModule;
```
Comment (cs-suyog-jinturkar) [inline widget/edit.controller.js:17]: loadFieldsOfModule or loadModuleFields?

File: widget/edit.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<!--Copyright start
+    MIT License
+    Copyright (c) 2024 Fortinet Inc
+Copyright end-->
+<!-- class="noMargin" name="editWidgetForm" data-ng-class="{'state-wait': processing } -->
+<form data-ng-submit="save()" name="editCyberThreatWorldMapForm">
+    <div class="modal-header">
+        <h3 class="modal-title col-md-9">Edit World Map</h3>
```
Comment (cs-suyog-jinturkar) [inline widget/edit.html:8]: title might need be i18n friendly

File: widget/edit.controller.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          sortedObj[key] = entity.fields[key];
+        }
+        $scope.countryField = sortedObj;
+
+        for (const key of Object.keys(sortedObj)) {
+          if (sortedObj[key].type === "picklist") {
+            filteredFieldsByPicklist[key] = sortedObj[key];
+          }
+        }
+        $scope.moduleField = filteredFieldsByPicklist;
```
Comment (cs-suyog-jinturkar) [inline widget/edit.controller.js:76]: $scope.moduleField --> $scope.fieldsByTypePicklist

File: widget/edit.controller.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      $scope.moduleField = [];
+      var entity = new Entity(config.worldMapModule);
+      entity.loadFields().then(function () {
+        const sortedKeys = Object.keys(entity.fields).sort();
+        const sortedObj = {};
+        const filteredFieldsByPicklist = {};
+        for (const key of sortedKeys) {
+          sortedObj[key] = entity.fields[key];
+        }
+        $scope.countryField = sortedObj;
```
Comment (cs-suyog-jinturkar) [inline widget/edit.controller.js:69]: it represents singlefield or list of fields? naming needs to be proper

can we use this same for loop to get picklist type of fields?

File: widget/info.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        "pages": [
+            "Dashboard"
+        ],
+        "certified": "Yes",
+        "publisher": "Fortinet",
+        "compatibility": [
+            "7.6.1"
+        ],
+        "category": [
+            "Information"
```
Comment (cs-suyog-jinturkar) [inline widget/info.json:19]: threat intel as well

File: widget/info.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    "name": "cyberThreatWorldMap",
+    "title": "Cyber Threat World Map",
+    "subTitle": "World Map widget which maps fields mapping to countries",
+    "version": "1.0.0",
+    "publishedDate": 1717391160,
+    "releaseNotes": "unavailable",
+    "metadata": {
+        "description": "",
+        "pages": [
+            "Dashboard"
```
Comment (cs-suyog-jinturkar) [inline widget/info.json:11]: make it available for all

File: widget/info.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "cyberThreatWorldMap",
+    "title": "Cyber Threat World Map",
+    "subTitle": "World Map widget which maps fields mapping to countries",
```
Comment (cs-suyog-jinturkar) [inline widget/info.json:4]: needs to be reworded

File: widget/view.controller.js [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+              countries.push(obj[countryField]);
+            }
+          }
+        }
+
+        const parentDiv = d3.select(`#world-map-${config.wid}`);
+        const widthP = parentDiv.node().getBoundingClientRect().width;
+        const heightP = parentDiv.node().getBoundingClientRect().height;
+
+        const width = 1100;
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:77]: hard coded?

File: widget/view.controller.js [hunk 9]
```diff
@@ -0,0 +1,385 @@
...
+        const height = 800;
+
+        const svg = d3.select(`#world-map-${config.wid}`).append("svg")
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:80]: what will happen with different themes?

File: widget/edit.controller.js (hunk 4 above)
Comment (cs-pranali-chavan) [reply widget/edit.controller.js:69]: it fetches countryField of. the module so its singular

**Review by cs-suyog-jinturkar — APPROVED** [review]

**Pull request merged**